src/rgw/rgw_reshard.cc

```diff
@@ -832,13 +834,17 @@ int RGWReshard::process_single_logshard(int logshard_num)
         ret = l.lock_exclusive(&store->reshard_pool_ctx, logshard_oid);
         if (ret == -EBUSY) { /* already locked by another processor */
           ldout(store->ctx(), 5) << __func__ << "(): failed to acquire lock on " << logshard_oid << dendl;
-          return ret;
+          throw ret;
```

Comment:
Looks better now. I think what you said before makes sense. Better don't unlock here, as we already got an `-EBUSY` when trying to lock.

Reply:
agreed.  already updated.